### PR TITLE
External commands exec changes

### PR DESCRIPTION
**Pull request by twe4ked**
Trying to fix [this issue](https://github.com/nushell/nushell/issues/347#issuecomment-524815077).

I haven't touched the Windows code but there are only minor differences between the two blocks now. The main one being the optional span (`let mut span = None;`).

I wasn't sure how to test all the different cases the code is trying to cover so there's a good chance I've missed something major.

New behaviour:

```
/Users/odin/Dev/github/wycats/nushell(external-commands-exec-changes)> help
error: Command not found
- shell:1:0
1 | help
  | ^^^^ not found
/Users/odin/Dev/github/wycats/nushell(external-commands-exec-changes)> sleep
usage: sleep seconds
/Users/odin/Dev/github/wycats/nushell(external-commands-exec-changes)> ls
━━━━┯━━━━━━━━━━━━━━━━━━━━┯━━━━━━━━━━━┯━━━━━━━━━━┯━━━━━━━━━━┯━━━━━━━━━━━━━━━┯━━━━━━━━━━━━━━━━┯━━━━━━━━━━━━━━━━
 #  │ name               │ type      │ readonly │ size     │ created       │ accessed       │ modified
────┼────────────────────┼───────────┼──────────┼──────────┼───────────────┼────────────────┼────────────────
  0 │ CODE_OF_CONDUCT.md │ File      │          │   3.4 KB │ an hour ago   │ an hour ago    │ an hour ago
````

### CODE REVIEW

File: src/commands/classified.rs [hunk 1]
```diff
@@ -340,19 +342,26 @@ impl ExternalCommand {
...
+                    if arg_chars.len() > 1
+                        && arg_chars[0] == '"'
+                        && arg_chars[arg_chars.len() - 1] == '"'
+                    {
+                        // quoted string
+                        let new_arg: String = arg_chars[1..arg_chars.len() - 1].iter().collect();
+                        process = process.arg(new_arg);
+                    } else {
+                        process = process.arg(arg.item.clone());
+                    }
```
Comment (twe4ked) [inline src/commands/classified.rs]: Copied from the windows version. There was a failing test around quoting.

**Comment by jonathandturner** [comment]
@twe4ked - does this need more testing before it's ready? I'm happy to try it out

**Comment by twe4ked** [comment]
Yeah, as I mentioned in the description I wasn't sure how to test some of the cases that the code looks like it's trying to handle. I'd love if you could give it a test, sorry about the half-finshed PR 😅.

**Comment by jonathandturner** [comment]
Sorry, I dropped the ball on this PR. I'm seeing some odd behavior in Windows that I'm also hoping to fix by a change to how we're doing subprocess, but I need to investigate further.

Should I close this PR, or should we adapt to trying to work on a more robust shell-out?

**Comment by twe4ked** [comment]
There's not much work here. I'll just close it. Thanks for taking a look!

**Pull request closed without merging**